pkg/appclient/jsonrpc.go

```diff
@@ -0,0 +1,116 @@
+package appclient
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/cometbft/cometbft/crypto"
+
+	"pocket/x/service/types"
+	sessiontypes "pocket/x/session/types"
+	sharedtypes "pocket/x/shared/types"
+)
+
+// handleJSONRPCRelays handles JSON RPC relay requests.
+func (app *appClient) handleJSONRPCRelays(
+	ctx context.Context,
+	serviceId string,
+	request *http.Request,
+) (responseBz []byte, err error) {
+	// Read the request body bytes.
+	payloadBz, err := io.ReadAll(request.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	// Hash and sign the request payload.
+	hash := crypto.Sha256(payloadBz)
+	signature, _, err := app.keyring.Sign(app.keyName, hash)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create the relay request payload.
+	relayRequestPayload := &types.RelayRequest_JsonRpcPayload{}
+	relayRequestPayload.JsonRpcPayload.Unmarshal(payloadBz)
+
+	// Get the current block height to query for the current session.
+	currentBlock := app.blockClient.LatestBlock(ctx)
+
+	// Query for the current session.
+	sessionQueryReq := sessiontypes.QueryGetSessionRequest{
+		ApplicationAddress: app.appAddress,
+		ServiceId:          &sharedtypes.ServiceId{Id: serviceId},
+		BlockHeight:        currentBlock.Height(),
+	}
+	sessionQueryRes, err := app.sessionQuerier.GetSession(ctx, &sessionQueryReq)
```

Comment:
As applications can be in a single session at a time we may want to look into a `SessionClient` similar to the `BlockClient` that removes the need to do this query on every request.

Reply:
I added a session cacher but didn't use the SessionClient. Feel free to change if you think it would be better